GEOSens_GridComp/GEOS_EnsGridComp.F90

```diff
@@ -2924,11 +2924,11 @@ subroutine Collect_land_ens(gc, import, export, clock, rc)
     VERIFY_(status)
     call MAPL_GetPointer(export, AVETSNOW_enavg,  'AVETSNOW' ,rc=status)
     VERIFY_(status)
-    call MAPL_GetPointer(export, FRSAT_enavg,  'FRSAT' ,rc=status)
+    call MAPL_GetPointer(export, FRSAT_enavg,  'FRSAT' , alloc=.true., rc=status)   ! always allocate to support ASNOW_enavg calc below
```

Comment:
@mathomp4, @weiyuan-jiang: 
Would it make sense to determine whether we really need to allocate  the FR* variables?  If the following works, it would save a small amount of memory at the expense of adding an if statement and a few extra lines of code.

```
if (associated(ASNOW_enavg)) then

  call MAPL_GetPointer(export, FRSAT_enavg,  'FRSAT' , alloc=.true., rc=status)   ! always allocate to support ASNOW_enavg calc below
  VERIFY_(status)
  call MAPL_GetPointer(export, FRUST_enavg,  'FRUST' , alloc=.true., rc=status)   ! always allocate to support ASNOW_enavg calc below
  VERIFY_(status)
  call MAPL_GetPointer(export, FRWLT_enavg,  'FRWLT' , alloc=.true., rc=status)   ! always allocate to support ASNOW_enavg calc below
  VERIFY_(status)

else

  call MAPL_GetPointer(export, FRSAT_enavg, 'FRSAT' ,rc=status)
  VERIFY_(status)
  call MAPL_GetPointer(export, FRUST_enavg,  'FRUST' ,rc=status)
  VERIFY_(status)
  call MAPL_GetPointer(export, FRWLT_enavg,  'FRWLT' ,rc=status)
  VERIFY_(status)

end if
```
This may not work and there's probably a more elegant way to make the allocation conditional.  And perhaps it's silly to worry about always or only sometimes allocating these three exports.  Thoughts?

Reply:
Well, one possibility (but would need testing) is first do the GetPointer for `ASNOW_enavg`. 

Then maybe this could work:

```

call MAPL_GetPointer(export, FRSAT_enavg,  'FRSAT' , alloc=associated(ASNOW_enavg), rc=status)

```

That might not though (maybe it's not kosher Fortran?). But you could have a logical and then:

```

logical :: ASNOW_enavg_associated
...

ASNOW_enavg_associated = associated(ASNOW_enavg)

call MAPL_GetPointer(export, FRSAT_enavg,  'FRSAT' , alloc=ASNOW_enavg_associated, rc=status)

```

Would want to test both just in case!

Reply:
Thanks, @mathomp4, for the hints.  Re. the suggestion:
```
ASNOW_enavg_associated = associated(ASNOW_enavg)
call MAPL_GetPointer(export, FRSAT_enavg,  'FRSAT' , alloc=ASNOW_enavg_associated, rc=status)
```
Would this still work if **only** FRSAT but **not** ASNOW is written?  Then we would have alloc=.false.  Not sure what MAPL_GetPointer() would do with the "alloc" spec when at the same time FRSAT output is requested in HISTORY.rc.

Reply:
@gmao-rreichle Hmm. you might be right. Staring at [the code](https://github.com/GEOS-ESM/MAPL/blob/df6786ca32f75b4848743301f5a31e6a6288622e/base/GetPointer.H#L78-L85), it seems like if `alloc` is present, it wins. So passing in `alloc=.false.` could maybe short circuit things.

I'm going to mention @bena-nasa and @atrayano here. They might know better, but my guess is the easy way is to just always `alloc` the needed fields or do the big if-block like you have up there.

Reply:
Actually, there are more dependence. ASNOW_enavg depends  on FRSAT_enavg, ..., and FRSAT_enavg depends on FRSAT in Catch_GridComp.  I think it is too tricky to enforce the dependence chain.  We can simply write out error message if ASNOW_enavg is requested but not those dependent variables.

Reply:
> Actually, there are more dependence. ASNOW_enavg depends on FRSAT_enavg, ..., and FRSAT_enavg depends on FRSAT in Catch_GridComp. I think it is too tricky to enforce the dependence chain. We can simply write out error message if ASNOW_enavg is requested but not those dependent variables.

Thanks, @weiyuan-jiang, for the input. Much appreciated.  I implemented a different solution, which I hope will work.  

@saraqzhang : When you get a chance, please try this branch to see whether you can now output just ASNOW (without also writing FRSAT, FRUST, and FRWLT.

Reply:
I tested the branch ( output ASNOW without FRSAT, FRUST, and FRWLT), it fixed the problem,  and the difference is on the order of 1e-7   if comparing ASNOW field from this output to ASNOW before ( dependent of  FRSAT, FRUST, and FRWLT).